src/wasm-binary.h

```diff
@@ -390,8 +390,8 @@ class WasmBinaryWriter : public WasmVisitor<WasmBinaryWriter, void> {
   void writeMemory() {
     if (wasm->memory.max == 0) return;
     if (debug) std::cerr << "== writeMemory" << std::endl;
-    o << int8_t(BinaryConsts::Memory) << int8_t(log2(wasm->memory.initial))
-                                      << int8_t(log2(wasm->memory.max))
+    o << int8_t(BinaryConsts::Memory) << int8_t(ceil(log2(wasm->memory.initial)))
+                                      << int8_t(ceil(log2(wasm->memory.max)))
```

Comment:
Why add `ceil`? When you use `std::log2()` you get the integral version but it returns a `double`? Aren't the integer values you care about all correctly represented by `double`?

Also, why `int8_t`?